components/tools/OmeroPy/test/integration/test_model51.py

```diff
@@ -100,3 +100,55 @@ def testAsMapMethod(self):
         )
         m = g.getConfigAsMap()
         assert m["foo"] == "bar"
+
+    def assertMapAnnotations(self, anns, length=2):
+        m = None
+        assert length == len(anns)
+        for idx in range(length):
+            a = anns[idx]
+            if isinstance(a, omero.model.MapAnnotationI):
+                m = anns[idx]
+        assert m
+        assert "foo" == m.getMapValue()[0].name
+        assert "bar" == m.getMapValue()[0].value
+
+    def testMapEagerFetch(self):
+        m = omero.model.MapAnnotationI()
+        m.setMapValue(
+            [NV("foo", "bar")]
+        )
+        m = self.update.saveAndReturnObject(m)
+        anns = self.query.findAllByQuery(
+            "select m from MapAnnotation m ",
+            None)
```

Comment:
See https://github.com/openmicroscopy/openmicroscopy/pull/3318/files#diff-cc72fc55d66d83492366c49e1420fff4R699 for instance. With the new `ITest` class, `self.client` will be shared by all test methods in the class. If the test has any assumption, it is possible to create `new_user_and_client` or `new_client` using the library methods and use this client instead.